### ISSUE TITLE
AbstractCollocationFinder from_documents() issues

### ISSUE THREAD

**Issue by fievelk** [nltk/nltk#1033]
If we pass a set of documents (e.g. tweets) to the `from_documents()` method we also get bigrams that never appeared in any document. This is due to `_itertools.chain(*documents)`, which collapses all the documents (so we end up with a single big document and we extract _its_ bigrams).

```
def from_documents(cls, documents):
    """Constructs a collocation finder given a collection of documents,
    each of which is a list (or iterable) of tokens.
    """
    return cls.from_words(_itertools.chain(*documents))
```

For example,using a list of two documents:

```
docs = [['I','am','a','farmer'], ['Newspapers','are','interesting']]

finder = BigramCollocationFinder.from_documents(docs)
finder.nbest(BigramAssocMeasures.raw_freq, 10)

>>> [('I', 'am'), ('Newspapers', 'are'), ('a', 'farmer'), ('am', 'a'), ('are', 'interesting'), ('farmer', 'Newspapers')]
```

So the method will also generate the bigram `('farmer', 'Newspapers')`, which actually never occurs in the documents. 

Con we consider it as an issue?

**Comment by jnothman** [comment]
Yes, it's an issue. A simple solution may involve putting n placeholders
between documents.

**Comment by stevenbird** [comment]
@longdt219 would you please implement @jnothman's suggestion.

**Comment by longdt219** [comment]
Sure @stevenbird,

**Comment by stevenbird** [comment]
@fievelk is this satisfactorily resolved now?

**Comment by fievelk** [comment]
@stevenbird I can confirm that the issue seems to be solved now:

```
>>> docs = [['I','am','a','farmer'], ['Newspapers','are','interesting']]

>>> finder = BigramCollocationFinder.from_documents(docs)
>>> finder.nbest(BigramAssocMeasures.raw_freq, 10)

[('I', 'am'), ('Newspapers', 'are'), ('a', 'farmer'), ('am', 'a'), ('are', 'interesting')]
```

The bigram `('farmer', 'Newspapers')` is now correctly omitted.

**Comment by alvations** [comment]
Resolved in #1086
